betty/job/__init__.py

```diff
@@ -0,0 +1,132 @@
+"""
+Provide utilities for running jobs concurrently.
+"""
+
+from __future__ import annotations
+
+from abc import abstractmethod, ABC
+from collections.abc import (
+    Awaitable,
+    Callable,
+    Mapping,
+)
+from typing import (
+    Any,
+    TYPE_CHECKING,
+    ParamSpec,
+    Generic,
+    Self,
+    Concatenate,
+    TypeAlias,
+    TypeVar,
+)
+
+from betty.job.context import Context
+
+if TYPE_CHECKING:
+    from types import TracebackType
+
+_JobP = ParamSpec("_JobP")
+
+_ContextT = TypeVar("_ContextT", bound=Context)
+
+
```

Comment:
Add an implementation that decorates one or more other runs, and aggregates the statistics.